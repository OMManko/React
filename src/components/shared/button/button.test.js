import { shallow } from 'enzyme';
import React from 'react';
import Button from './button';

describe('<Button />', () => {
    it('should render button and match snapshot', () => {
        const component = shallow(<Button />);
        expect(component).toMatchSnapshot();
    });

    it('should render button with class', () => {
        const variant = 'test';
        const component = shallow(<Button variant={variant}/>);
        expect(component).toMatchSnapshot();
        expect(component.props().className).toEqual(variant);
    });

    it('should render button with label', () => {
        const label = 'test';
        const component = shallow(<Button label={label}/>);
        expect(component).toMatchSnapshot();
        expect(component.find('button').text()).toMatch(label);
    });

    it('should call function when button is clicked', () => {
        const mockFn = jest.fn();
        const component = shallow(<Button handleAction={mockFn} />);
        component.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });
});
