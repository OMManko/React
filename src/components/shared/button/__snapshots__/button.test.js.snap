// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button and match snapshot 1`] = `ShallowWrapper {}`;

exports[`<Button /> should render button with class 1`] = `ShallowWrapper {}`;

exports[`<Button /> should render button with label 1`] = `ShallowWrapper {}`;
